cmake/tests.cmake

```diff
@@ -13,3 +13,30 @@ function(register_test)
       "$<TARGET_FILE:${TEST_TARGET}>"
       "--gtest_output=xml:${TESTRESULTS_DIRECTORY}/${TEST_NAME}_sponge_log.xml")
 endfunction()
+
+if(NOT TARGET GTest::GTest)
+        add_library(GTest::GTest INTERFACE IMPORTED)
+  target_link_libraries(GTest::GTest INTERFACE CONAN_PKG::gtest)
+endif()
+
+if(NOT TARGET GTest::Main)
+  set(FILE_PATH ${CMAKE_BINARY_DIR}/test_main.cpp)
+  if(NOT EXISTS ${FILE_PATH})
+    set(GTEST_MAIN_CPP "\
+#include <cstdio>\n\
+#include <gtest/gtest.h>\n\
+\n\
+int main(int argc, char* argv[]) {\n\
+  printf(\"Running main() from %s\\n\", __FILE__)\;\n\
+  ::testing::InitGoogleTest(&argc, argv)\;\n\
+  return RUN_ALL_TESTS()\;\n\
+}\n\
+")
+
```

Comment:
I am just curious why all this is necessary? I bet there is a good reason

Reply:
I tried to describe that in the PR description. It's needed because the conan-generated cmake config does not support linking to some part of a package. (A conan feature which is currently in development and will be merged soon.) So we either have to globally enable linking against gtest with an integrated main function or provide or own main function and link against that in the cases where it's needed. Since we have one test case which provides their own main function in OrbitSshQt, I had to choose the latter.

Reply:
That sounds a lot like a good reason for not moving to conan-generated cmake-config files. But doing that will allow us to fully support visual studio as soon as we moved all packages to conan-generated cmake-config files. Qt and GRPC are still missing.

Reply:
tbh, I should have read the PR description more closely. Thank you for taking the time. The SshQt integration test was not on the top of my mind anymore. It all makes sense now